### ISSUE TITLE
Confusing UX when prompting invited external users to log in

### ISSUE THREAD

**Issue by tim-goddard-flexcompute** [tailscale/tailscale#13062]
### What is the issue?

I've had a repeated pattern when inviting external users to join our Tailnet, that they get confused after following the invitation link, because they attempt to enter their email address in to the page they're taken to. They then come to me for help, and have to be prompted to choose an SSO method instead (this seems to work).

I think the issue here is that an option is presented first to type an email address rather than select an SSO method, where this is only valid for an existing account. This confuses external users invited to an account, who do not otherwise use Tailscale.

The UI could more actively guide the external user to prefer SSO, or make it clearer that the option to log in by typing an email address is only valid for existing users.

### Steps to reproduce

The workflow looks like:

- I send an invitation by email.
- Invitee opens the email, and clicks the contained link.
- They enter their email address.
- They get the following error message:

![invitee-login](https://github.com/user-attachments/assets/3b61f6d1-cf2a-4fb1-9884-b1686f1de986)

### Are there any recent changes that introduced the issue?

N/A

### OS

Linux, macOS, Windows

### Bug report

BUG-de1b56ed6e39aeafb1522d32c01517b71b1837a6be77bcff43483698bcc69c01-20240808023833Z-05d05b5b93c958b9

**Comment by markhu** [comment]
I also was confused by this. Tailscale’s invite flow can be confusing because it assumes the invited user already has a Tailscale account. If they don’t, it often defaults to creating a new Tailnet instead of joining the intended shared one. We developed this workaround: Log out of Tailscel, e.g. https://login.tailscale.com/logout then log back in, and only then are they presented with the choice of their own Tailnet, or the one they were invited to.

**Comment by altano** [comment]
Another point of confusion is that when they get into their new tailnet, having come from the invite link, the invite flow is NOT resumed. The new user is just presented with an interstitial asking them to add new devices to their new tailnet (which has nothing to do with the invite). Closing that window and clicking the invite link again will always take you to this interstitial.

In the interstitial is a "skip" link. When you click that, you're taken to your account home. NOW if you click the invite link, it works.

In short, the invite flow, if accepted by a new user, is pretty broken atm. Really confusing and bad UX.
